src/zenml/config/retry_config.py

```diff
@@ -0,0 +1,24 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Retry configuration for a step."""
+
+from zenml.config.strict_base_model import StrictBaseModel
+
+
+class StepRetryConfig(StrictBaseModel):
+    """Retry configuration for a step."""
```

Comment:
Maybe you should specify the units for the delay and backoff. is this hours, minutes, seconds? (so add that to the docstring and maybe make some mention of this in the docs, too?

Reply:
already specified in docs

Reply:
But just add it in the docstring? Allows someone to see that from the mouseover in their IDE instead of having to go look in the docs.